### ISSUE TITLE
scvelo.pl.velocity_graph.py: ValueError: inhomogenous array

### ISSUE THREAD

**Issue by jwalewski** [theislab/scvelo#1121]
Hello, I am attempting to perform RNA velocity on a blood sample previously run through cellranger and velocyto (so scVelo's input was a loom file). After attempting to follow the standard vignette, I got an error from velocity_graph.py - more specfically, when trying to graph the velocities and compute the cosine similarities (and even more specifically when trying to parallelize this work).

```
#0. Import sys
import sys
#1. Import scvelo
import scvelo
#2. Configure scvelo to support visualizations
scvelo.set_figure_params()
#3. read input data from loom file generated from velocyto
adata = scvelo.read("[PATH TO MY FILES]/200427_Blood_GEX.loom", cache=True)
print("Step 3 suceeded")
#4. Filter and preprocessing
scvelo.pp.filter_and_normalize(adata, flavor="cell_ranger") #first test with my parameters; I figure since this data originally came from cellranger this should be its flavor
scvelo.pp.moments(adata) #no other parameters needed IMO
#5. RNA Velocity
print("step 4 completed successfully")
scvelo.tl.velocity(adata, mode="stochastic") #mode refers to which model to use - 3 options, tradeoff between computational load and accuracy
print("Step 4A Completed Successfully")
print("this is adata.X: ", adata.X)
print("this is adata..shape[0]: ", adata.X.shape[0])
print("this is adata..shape[1]: ", adata.X.shape[1])
scvelo.tl.velocity_graph(adata) #no other parameters needed IMO
**#step 4b here is the step which is failing.** 
#6. Visualization - Most of these parameters are purely comsmetic preference
scvelo.pl.velocity_embedding(adata, basis="umap") #no other parameters needed, the other option for this one is tsne (umap is recommended in vignettes)
scvelo.pl.velocity_embedding_grid(adata, basis="umap") #parameters on page 47 of documentation (https://scvelo.readthedocs.io/_/downloads/en/stable/pdf/)
scvelo.pl.velocity_embedding_stream(adata, basis="umap") 
#7. Extras
scvelo.pl.velocity(adata)
scvelo.pl.velocity_graph(adata)
sys.exit()
```

Here's the exact error:

```
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (1, 4) + inhomogeneous part.
```
Any suggestions on how to approach this?

**Comment by Chrisfang11** [comment]
same problem

**Comment by ninning** [comment]
I ended up creating a conda environment and set the python install to version 3.6
`conda create --name scvelo python=3.6`
Since that seems to be the version that scvelo was initially developed in and that seemed to have fixed the issue.

It probably is due to the numpy version as this environment installed numpy 1.19.5 for me instead of the  1.24 + versions that the other environment did.

I just tested it on the demo data shown below.

![Screenshot 2023-10-03 at 3 02 19 PM](https://github.com/theislab/scvelo/assets/24235120/72a7b734-0600-431e-9e8e-915c25ae2f16)

**Comment by ramadatta** [comment]
I used the following commands and created a new environment based on this [comment](https://github.com/theislab/scvelo/issues/1040#issuecomment-1556059697).

conda create -n scvelo_env python=3.9
conda activate scvelo_env
conda install numpy scipy cython numba matplotlib scikit-learn h5py click
pip install notebook
pip install scanpy
pip install leidenalg
pip install scvelo
pip install pandas==1.1.5 
pip install numpy==1.21.1

This error is resolved for me.

**Comment by WeilerP** [comment]
Fixed in #1059.
